### PR TITLE
[SMTChecker] Update test expectations for z3 4.8.6

### PR DESCRIPTION
**Pull request by leonardoalt**
Should be merged after the new docker image is pushed

### CODE REVIEW

**Comment by ekpyron** [comment]
This should also update the osx install deps script and add it to the circleci cache key (see https://github.com/ethereum/solidity/pull/7464#discussion_r344735820)

File: .circleci/osx_install_dependencies.sh [hunk 1]
```diff
@@ -43,13 +43,13 @@ then
   ./scripts/install_obsolete_jsoncpp_1_7_4.sh
 
   # z3
```
Comment (ekpyron) [inline .circleci/osx_install_dependencies.sh:45]: Probably all of this could be replaced by just ``brew install z3`` now. However, then we'd need a different criterion for the outer ``if`` and maybe better to do all this in a separate PR afterwards.

**Review by ekpyron — APPROVED** [review]
Fine with me once the tests pass! Although we might want to have another followup PR for using brew's z3 again on os x.

File: .circleci/osx_install_dependencies.sh (hunk 1 above)
Comment (leonardoalt) [reply .circleci/osx_install_dependencies.sh:45]: Yea, agree

**Comment by ekpyron** [comment]
Actually: since the clang test run is set as optional and the respective docker image seems to be delayed in its update, I will merge this despite the failing tests.
